### PR TITLE
Update rebuild flag to true when images are updated

### PR DESCRIPTION
**Pull request by NehaNaithani**
- Does the iofog.org or README.md documentation need to change?
  - [ ] Yes 
  - [ ] No

### CODE REVIEW

File: src/services/microservices-service.js [hunk 1]
```diff
@@ -437,7 +443,7 @@ async function updateMicroserviceEndPoint (microserviceUuid, microserviceData, u
   } else if (!microservice.catalogItemId && microserviceDataUpdate.images && microserviceDataUpdate.images.length === 0) {
     // No catalog, and no image
     throw new Errors.ValidationError(AppHelper.formatMessage(ErrorMessages.MICROSERVICE_DOES_NOT_HAVE_IMAGES, microserviceData.name))
-  } else if (microserviceDataUpdate.images && microserviceDataUpdate.images.length > 0) {
+  } else if (!checkIfMicorserviceImageIsUpdated(microserviceDataUpdate.images, microserviceImages)) {
```
Comment (Pixcell) [inline src/services/microservices-service.js]: typo in the name ;)

File: src/services/microservices-service.js [hunk 2]
```diff
@@ -530,6 +535,24 @@ async function updateMicroserviceEndPoint (microserviceUuid, microserviceData, u
   await _updateChangeTracking(true, updatedMicroservice.iofogUuid, transaction)
 }
 
+/**
+ * checks if microservice image is updated
+ * @param {*} microserviceDataUpdateImages
+ * @param {*} catalogImages
+ */
+const checkIfMicorserviceImageIsUpdated = (microserviceDataUpdateImages, catalogImages) => {
```
Comment (Pixcell) [inline src/services/microservices-service.js]: The name is wrong, should be checkIfMicroserviceImageIsEqual , there it looks like the function will return true if they are different

**Review by Pixcell — CHANGES_REQUESTED** [review]

File: src/services/microservices-service.js (hunk 2 above)
Comment (NehaNaithani) [reply src/services/microservices-service.js]: updated

File: src/services/microservices-service.js (hunk 1 above)
Comment (NehaNaithani) [reply src/services/microservices-service.js]: updated

**Pull request merged**